core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -1106,6 +1112,10 @@ class Partition(val topicPartition: TopicPartition,
    * @return true if the HW was incremented, and false otherwise.
    */
   private def maybeIncrementLeaderHW(leaderLog: UnifiedLog, currentTimeMs: Long = time.milliseconds): Boolean = {
+    if (isUnderMinIsr) {
+      trace(s"Skip checking whether HWM can advance because partition is under min ISR(ISR=${partitionState.isr}")
```

Comment:
Should we word this more like "Not increasing HWM..." so it's explicit that the HWM is not increasing?

Reply:
Done.